File: src/airco_demo.cpp
#include <iostream>
#include <memory>
#include "airco.h"

int main()
{
    std::cout << "Testing Airco class" << std::endl;

    std::cout << "Creating Airco" << std::endl;
    std::shared_ptr<Airco> airco = std::make_shared<Airco>();
    std::cout << "Initial airco state: " << airco->GetState() << std::endl;

    std::cout << "Registering callback function" << std::endl;
    auto f = [](Airco::State old_state, Airco::State new_state) {
        std::cout << "Transitioning from " << old_state << " to " << new_state << std::endl;
    };
    airco->AddCallback(f);
    std::cout << "Switching airco on" << std::endl;
    airco->On();
    std::cout << "Switching airco off" << std::endl;
    airco->Off();
    std::cout << "No callback for repeatedly setting the same state" << std::endl;
    airco->Off();
}

File: src/airco_demo.py
from airco import Airco


if __name__ == '__main__':
    print("Testing airco")
    a = Airco()
    print(f"After creation, airco state is {a.state.name}")

    print("Adding callback functions")
    f1 = lambda o_s, n_s: print(f"Transitioning from {o_s.name} to {n_s.name}")
    f2 = lambda o_s, n_s: print(f"I said.. {o_s.name} to {n_s.name}")
    a.AddCallback(f1)
    a.AddCallback(f2)

    a.On()
    a.Off()

